### PR TITLE
Base entity

### PR DESCRIPTION
**Pull request by KBerndt10**
- Created Base entity class
- refactored Player to implement Base entity class
- updated gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,23 +1,27 @@
...
 /[Bb]uilds/
 /[Ll]ogs/
+/[Uu]ser[Ss]ettings/
+
+# MemoryCaptures can get excessive in size.
+# They also could contain extremely sensitive data
 /[Mm]emoryCaptures/
 
-# Asset meta data should only be ignored when the corresponding asset is also ignored
-!/[Aa]ssets/**/*.meta
```
Comment (ryiuthere) [inline .gitignore:14]: Any particular reason why this is being removed? Seems nice to have.

File: Assets/Scripts/BaseEntity.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public void OnHit(int damage) {
+        // @TODO: finish once projectils exist
+        this.health = Math.Max(0, this.health - damage);
+        if (this.health <= 0) {
+            OnDeath();
+        }
+    }
+
+    /** Hook to modify Start behavior */
+    public void StartHook() {}
```
Comment (ryiuthere) [inline Assets/Scripts/BaseEntity.cs]: If this is to modify start behavior, should this be virtual?

File: .gitignore (hunk 1 above)
Comment (KBerndt10) [reply .gitignore:14]: I just copied the latest version from the unity ignore repo you got the original version from. I can put this back, but it won't actually make a difference likely.

File: Assets/Scripts/BaseEntity.cs (hunk 2 above)
Comment (KBerndt10) [reply Assets/Scripts/BaseEntity.cs]: Good catch, I'll add that

**Comment by KBerndt10** [comment]
@ryiuthere 
Your comments have been resolved in the latest commit

**Review by ryiuthere — APPROVED** [review]

**Pull request merged**